internal/manifest/manifest.go

```diff
@@ -0,0 +1,194 @@
+package manifest
+
+import (
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/klauspost/compress/zstd"
+
+	"github.com/canonical/chisel/internal/jsonwall"
+	"github.com/canonical/chisel/internal/setup"
+)
+
+const schema = "1.0"
+
+type Package struct {
+	Kind    string `json:"kind"`
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+	Digest  string `json:"sha256,omitempty"`
+	Arch    string `json:"arch,omitempty"`
+}
+
+type Slice struct {
+	Kind string `json:"kind"`
+	Name string `json:"name,omitempty"`
+}
+
+type Path struct {
+	Kind      string   `json:"kind"`
+	Path      string   `json:"path,omitempty"`
+	Mode      string   `json:"mode,omitempty"`
+	Slices    []string `json:"slices,omitempty"`
+	Hash      string   `json:"sha256,omitempty"`
+	FinalHash string   `json:"final_sha256,omitempty"`
+	Size      uint64   `json:"size,omitempty"`
+	Link      string   `json:"link,omitempty"`
+}
+
+type Content struct {
+	Kind  string `json:"kind"`
+	Slice string `json:"slice,omitempty"`
+	Path  string `json:"path,omitempty"`
+}
+
+type Manifest struct {
+	db *jsonwall.DB
+}
+
+// Read loads a Manifest from a file without performing any validation. The file
+// is assumed to be both valid jsonwall and a valid Manifest (see [Validate]).
+func Read(absPath string) (manifest *Manifest, err error) {
```

Comment:
I've missed this on the first review: why is this not an io.Reader?  It seems curious that it's not only a filename, but a filename to a zstd compressed file.  Doesn't feel like this package should care about the compression format of the manifest, or how the manifest is being read from disk. Note how the jsonwall.ReadDB function blended nicely below.  We won't be able to do this with this function.

Reply:
Agree I was thinking that zstd was part of the "format" but there is nothing dictating that, we could be storing it in a completely different way and this package should continue working correctly.

I have made the changes in https://github.com/canonical/chisel/pull/144/files/56dd49ca33944ab4e09cc7778cdaec7245732924..4e0e33b13cd8319e31fb0a0c36587cfa479dfd7c.